### PR TITLE
Add doc for rewriting manifest with spec id

### PR DESCRIPTION
**Pull request by puchengy**
Doc for new feature: https://github.com/apache/iceberg/pull/9242 and https://github.com/apache/iceberg/pull/9243.

### CODE REVIEW

File: docs/spark-procedures.md [hunk 1]
```diff
@@ -446,6 +446,7 @@ This procedure invalidates all cached Spark plans that reference the affected ta
 |---------------|-----------|------|-------------|
 | `table`       | ✔️  | string | Name of the table to update |
 | `use_caching` | ️   | boolean | Use Spark caching during operation (defaults to true) |
+| `spec_id`     | | int | Spec id of the manifests to rewrite (defaults to latest spec id) |
```
Comment (aokolnychyi) [inline docs/spark-procedures.md:449]: I think `latest` should be `current`

**Comment by puchengy** [comment]
@aokolnychyi Addressed your comment, PTAL, thanks!

File: docs/spark-procedures.md [hunk 2]
```diff
@@ -442,10 +442,11 @@ This procedure invalidates all cached Spark plans that reference the affected ta
...
 
-| Argument Name | Required? | Type | Description |
-|---------------|-----------|------|-------------|
-| `table`       | ✔️  | string | Name of the table to update |
-| `use_caching` | ️   | boolean | Use Spark caching during operation (defaults to true) |
+| Argument Name | Required? | Type | Description                                                   |
+|---------------|-----------|------|---------------------------------------------------------------|
+| `table`       | ✔️  | string | Name of the table to update                                   |
+| `use_caching` | ️   | boolean | Use Spark caching during operation (defaults to true)         |
+| `spec_id`     | | int | Spec id of the manifests to rewrite (defaults to current spec id) |
```
Comment (ajantha-bhat) [inline docs/spark-procedures.md:449]: nit: we can have formatting as above line

```suggestion
| `spec_id`     | ️   | int | Spec id of the manifests to rewrite (defaults to current spec id) |
```

**Review by ajantha-bhat — APPROVED** [review]

**Review by Fokko — APPROVED** [review]
Thanks @puchengy for adding this to the docs, and @ajantha-bhat @aokolnychyi for the reviews!

**Pull request merged**